ksql-api/src/main/java/io/confluent/ksql/api/plugin/KsqlServerEndpoints.java

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.api.plugin;
+
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+import io.confluent.ksql.api.impl.Utils;
+import io.confluent.ksql.api.server.BaseServerEndpoints;
+import io.confluent.ksql.api.server.PushQueryHandler;
+import io.confluent.ksql.api.spi.InsertsSubscriber;
+import io.confluent.ksql.engine.KsqlEngine;
+import io.confluent.ksql.parser.KsqlParser.ParsedStatement;
+import io.confluent.ksql.parser.KsqlParser.PreparedStatement;
+import io.confluent.ksql.parser.tree.Query;
+import io.confluent.ksql.parser.tree.Statement;
+import io.confluent.ksql.query.RowConsumer;
+import io.confluent.ksql.schema.ksql.Column;
+import io.confluent.ksql.schema.ksql.FormatOptions;
+import io.confluent.ksql.schema.ksql.LogicalSchema;
+import io.confluent.ksql.schema.registry.KsqlSchemaRegistryClientFactory;
+import io.confluent.ksql.security.KsqlSecurityExtension;
+import io.confluent.ksql.services.ServiceContext;
+import io.confluent.ksql.statement.ConfiguredStatement;
+import io.confluent.ksql.util.KsqlConfig;
+import io.confluent.ksql.util.KsqlStatementException;
+import io.confluent.ksql.util.QueryMetadata;
+import io.vertx.core.Context;
+import io.vertx.core.WorkerExecutor;
+import io.vertx.core.json.JsonObject;
+import java.security.Principal;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.OptionalInt;
+import java.util.function.Supplier;
+import org.apache.kafka.streams.KafkaClientSupplier;
+import org.apache.kafka.streams.processor.internals.DefaultKafkaClientSupplier;
+import org.reactivestreams.Subscriber;
+
+public class KsqlServerEndpoints extends BaseServerEndpoints {
+
+  private final KsqlEngine ksqlEngine;
+  private final KsqlConfig ksqlConfig;
+  private final KsqlSecurityExtension securityExtension;
+  private final ServiceContextFactory theServiceContextFactory;
```

Comment:
nit: why not simply `serviceContextFactory`? (I assume there's a reason for diverging from convention -- I'm just not seeing what it is.)

Reply:
It's vestigial from when there was something else called serviceContextFactory